File: DotNet Project V2.0/backend-dotnet/src/Emart.Api/Controllers/UserController.cs
using Emart.Application.Dtos;
using Emart.Application.Services.Interfaces;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Cors;
using Microsoft.AspNetCore.Mvc;

namespace Emart.Api.Controllers;

// SecurityConfig: GET /api/users -> ADMIN, DELETE /api/users/** -> ADMIN, everything else under
// /api/users/** (including POST create and PUT update) -> any authenticated user.
[ApiController]
[Route("api/users")]
[EnableCors("Frontend")]
[Authorize]
public class UserController : ControllerBase
{
    private readonly IUserService _userService;

    public UserController(IUserService userService)
    {
        _userService = userService;
    }

    [HttpPost]
    public async Task<ActionResult<UserResponseDTO>> CreateUser([FromBody] UserRequestDTO request, CancellationToken ct) =>
        Ok(await _userService.SaveUserAsync(request, ct));

    [HttpGet]
    [Authorize(Roles = "ADMIN")]
    public async Task<ActionResult<IReadOnlyList<UserResponseDTO>>> GetAllUsers(CancellationToken ct) =>
        Ok(await _userService.GetAllUsersAsync(ct));

    [HttpGet("{id:long}")]
    public async Task<ActionResult<UserResponseDTO>> GetUserById(long id, CancellationToken ct) =>
        Ok(await _userService.GetUserByIdAsync(id, ct));

    [HttpPut("{id:long}")]
    public async Task<ActionResult<UserResponseDTO>> UpdateUser(long id, [FromBody] UserRequestDTO request, CancellationToken ct) =>
        Ok(await _userService.UpdateUserAsync(id, request, ct));

    [HttpDelete("{id:long}")]
    [Authorize(Roles = "ADMIN")]
    public async Task<ActionResult<string>> DeleteUser(long id, CancellationToken ct)
    {
        await _userService.DeleteUserAsync(id, ct);
        return Ok("User deleted successfully.");
    }
}

File: Java Project  V1.7/backend-spring/src/main/java/com/emart/controller/UserController.java
package com.emart.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import com.emart.dto.UserRequestDTO;
import com.emart.dto.UserResponseDTO;
import com.emart.service.UserService;

@RestController
@RequestMapping("/api/users")
@CrossOrigin(origins = "http://localhost:5173")
public class UserController {

    @Autowired
    private UserService userService;

    // Create User
    @PostMapping
    public UserResponseDTO createUser(@RequestBody UserRequestDTO userRequestDTO) {
        return userService.saveUser(userRequestDTO);
    }

    // Get All Users
    @GetMapping
    public List<UserResponseDTO> getAllUsers() {
        return userService.getAllUsers();
    }

    // Get User By ID
    @GetMapping("/{id}")
    public UserResponseDTO getUserById(@PathVariable Long id) {
        return userService.getUserById(id);
    }

    // Update User
    @PutMapping("/{id}")
    public UserResponseDTO updateUser(@PathVariable Long id,
                                      @RequestBody UserRequestDTO userRequestDTO) {
        return userService.updateUser(id, userRequestDTO);
    }

    // Delete User
    @DeleteMapping("/{id}")
    public String deleteUser(@PathVariable Long id) {

        userService.deleteUser(id);

        return "User deleted successfully.";
    }

}